app/plugins/Wallet/js/lifecycle.js

```diff
@@ -91,14 +91,17 @@ function start() {
 	// DEVTOOL: uncomment to bring up devtools on plugin view
 	// IPC.sendToHost('devtools');
 	
-	// Need to check if wallet's unencrypted
-	IPC.sendToHost('api-call', '/wallet', 'on-opened');
+	// Need to check if wallet's unencrypted, keep checking until we get a response
+	updating = setInterval(IPC.sendToHost, 500, 'api-call', '/wallet', 'on-opened');
```

Comment:
Actually, it makes more sense to just combine the 'on-opened' and 'update-status' listeners. They're the same call and data, so why not